tasks/main.yml

```diff
@@ -3,48 +3,56 @@
 ---
 # tasks file for telegraf-docker-in-docker
 
-- name: Ensure system user is present on the host and in group docker
-  ansible.builtin.user:
-    name: "{{ tdid_user }}"
-    system: true
-    append: true
-    groups: docker
+- name: Run all tasks for role telegraf-docker-in-docker
+  when: tdid_arch_is_supported
+  block:
```

Comment:
This is a very long block. It may be beneficial to move the tasks to a separate file and include it here with the when-condition for readability.

Reply:
This would mean one very short main.yml with two conditional tasks, one calling into all the tasks of another file and the other for the informational message. The other file would have the same content as the former main.yml ... well why not.

Reply:
Yes, one file would be very short. But then one file only deals with the decision and the other file has the business logic. It's clean code. :)